### PR TITLE
Add new integration for YandexGPT Embedding Model

### PR DESCRIPTION
**Pull request by KirillKukharev**
# Description

This pull request introduces a new class YandexGPTEmbedding for generating embeddings using the Yandex Cloud API.

Fixes # (issue)

## New Package?

- [x] Yes
- [ ] No

## Version Bump?

- [x] Yes
- [ ] No

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/README.md
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/README.md:1]: Do we want a more detailed readme here? It shows up on llamahub.ai

File: llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        super().__init__(
+            model_name=model_name,
+            api_key=api_key,
+            folder_id=folder_id,
+            embed_batch_size=embed_batch_size,
+            callback_manager=callback_manager,
+            **kwargs,
+        )
+
+    def _getModelUri(self, is_document=False) -> str:
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py:86]: Can we please type the functions. `def _getModelUri(self, is_document: bool = False) -> str:`

File: llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/util.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class YException(Exception):
+    """Custom exception for YAuth related errors."""
+
+
+class YAuth:
+    """Class to handle authentication using folder ID and API key."""
+
+    def __init__(self, folder_id, api_key=None, iam_token=None) -> None:
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/util.py:12]: Similar comment here, nothing is typed

Comment (KirillKukharev) [reply llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/util.py]: Hi, I have updated my pull request according to your remarks. Could you please review it again at your earliest convenience?
Thank you for your time and feedback!

File: llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+check-filenames = true
+check-hidden = true
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
+[tool.llamahub]
+contains_example = true
+import_path = "llama_index.embeddings.yandexgpt"
+
+[tool.llamahub.class_authors]
+VoyageEmbedding = "llama-index"
```
Comment (nerdai) [inline llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/pyproject.toml:15]: Should be `YandexGPTEmbedding` and feel free to replace `llama-index` with your github username.

**Review by nerdai — COMMENTED** [review]
Thanks @KirillKukharev.

I pushed a commit to pass our lint checks. Left also a minor comment wrt `pyproject.toml`

File: llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                                f"No embedding found, result returned: {result}"
+                            )
+        except RetryError:
+            raise YException(
+                f"Error computing embeddings after {self.retries} retries. Result returned:\n{result}"
+            )
+
+    def _get_query_embedding(self, text: str) -> List[float]:
+        """Get query embedding sync."""
+        return self._embed(text, is_document=True)
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py:176]: Shouldn't this be is_document=False?

File: llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+    def _get_query_embeddings(self, texts: List[str]) -> List[List[float]]:
+        """Get list of queries embeddings sync."""
+        embeddings = []
+        for text in texts:
+            embeddings.append(self._embed(text, is_document=True))
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py:182]: same here?

File: llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py (hunk 4 above)
Comment (logan-markewich) [reply llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py:176]: (maybe I misunderstand this option though, but the query is not a document)

File: llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py (hunk 5 above)
Comment (KirillKukharev) [reply llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py:182]: > same here?

Oh, I'm sorry, I didn't fully understand the difference between text and query. According to docs clarifai, embedding methods should be named vice versa. I have swapped the names of the embedding methods, please see if it's ok?

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**